### PR TITLE
Implemented TakeWhile

### PR DESCRIPTION
**Pull request by nikolalukovic**

### CODE REVIEW

File: test/unitary/IQueryable.test.ts [hunk 1]
```diff
@@ -1286,6 +1306,61 @@ export namespace IQueryableUnitTest
...
+        it("Empty if empty (false) (iterator)", () =>
+        {
+            const base = instancer([]);
+            Test.isArrayEqual(new Enumerable(base.takeWhile(e => false)).toArray(), []);
+        });
+
+        it("Value is correct (returns elements)", () =>
+        {
+            const base = instancer([39, 40, 21, 66, 20]);
+            const takeWhile = base.takeWhile(e => e >= 39);
```
Comment (isc30) [inline test/unitary/IQueryable.test.ts]: shadowing function name, give it another name

Comment (isc30) [reply test/unitary/IQueryable.test.ts]: on all of them

Comment (nikolalukovic) [reply test/unitary/IQueryable.test.ts]: i'm not sure why it reports shadowing since it's identical code to skipWhile, where, etc... hmmm

Comment (isc30) [reply test/unitary/IQueryable.test.ts]: hmmmm that's strange, yep
nevermind, the PR is fine

**Pull request merged**